string state
uint32 error_log